### PR TITLE
Support for `StepRange`s using `ZonedDateTime`s

### LINKED ISSUE #20: StepRanges that use ZonedDateTimes fail
Julia doesn't like to create ranges using `a:b:c` syntax when `a` and `c` are `ZonedDateTime`s.

```
julia> tz = TimeZone("America/Winnipeg")
America/Winnipeg

julia> ZonedDateTime(DateTime(2016, 3, 13), tz):Hour(1):ZonedDateTime(DateTime(2016, 3, 14), tz)
ERROR: MethodError: no method matching guess(::TimeZones.ZonedDateTime, ::TimeZones.ZonedDateTime, ::Base.Dates.Hour)
Closest candidates are:
  guess(::DateTime, ::DateTime, ::Any)
  guess(::Date, ::Date, ::Any)
 in len(::TimeZones.ZonedDateTime, ::TimeZones.ZonedDateTime, ::Base.Dates.Hour) at ./dates/ranges.jl:13
 in steprem(::TimeZones.ZonedDateTime, ::TimeZones.ZonedDateTime, ::Base.Dates.Hour) at ./dates/ranges.jl:25
 in steprange_last(::TimeZones.ZonedDateTime, ::Base.Dates.Hour, ::TimeZones.ZonedDateTime) at ./range.jl:55
 in colon(::TimeZones.ZonedDateTime, ::Base.Dates.Hour, ::TimeZones.ZonedDateTime) at ./range.jl:100
 in eval(::Module, ::Any) at ./boot.jl:267
```

This can be fixed by implementing `guess(::ZonedDateTime, ::ZonedDateTime, ::Any)`, which is a fairly trivial function (I'm not incredibly fond of its *name*, but there you go). I'm working on a PR for this right now (just polishing up the test cases, actually), and it will allow for ranges do smart things like ignore nonexistent hours for DST transitions.

### PR DESCRIPTION
**Pull request by spurll**
Added `guess(::ZonedDateTime, ::ZonedDateTime, ::Any)`, which allows a `StepRange` to be constructed with `ZonedDateTime`s. Also added appropriate test cases and documentation. Closes #20.

### CODE REVIEW

File: src/timezones/ranges.jl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Base.Dates: guess, toms
+
+"""
+Given a start and end date, indicates how many steps/periods are between them. Defining this
+function allows `StepRange`s to be defined for `ZonedDateTime`s.
+"""
+function guess(start::ZonedDateTime, finish::ZonedDateTime, step)
+    floor(Int64, (Int128(finish.utc_datetime) - Int128(start.utc_datetime)) / toms(step))
```
Comment (omus) [inline src/timezones/ranges.jl]: Probably should just be using `Dates.guess(::DateTime, ::DateTime, ::Any)` and change this to be `guess(start.utc_datetime, finish.utc_datetime, step)`

Comment (spurll) [inline src/timezones/ranges.jl]: Good call. Stand by.

**Comment by omus** [comment]
@spurll your PR looks good. I am going to try to fix the 0.5 issues prior to merging this.

**Comment by omus** [comment]
@spurll could you rebase your changes onto the latest master? The pre-existing 0.5 issues have been mitigated.

**Pull request merged**